### PR TITLE
Added display of humidity to Today section, along with configuration

### LINKED ISSUE #273: Humidity
**Is your feature request related to a problem? Please describe.**
It would be nice if it's possible to display the humidity from an specific entity (like `temperature_sensor`)

**Describe the solution you'd like**
That it would be possible to use `humidity_sensor` and that it is next (right side) the temperature

**Describe alternatives you've considered**
Different card, not possible either.

### PR DESCRIPTION
**Pull request by rakasha681**
Resolves #273 

Added display of humidity to Today section below weather/temp, along with configuration for sensor and option hide_humidity. Updated readme to match latest changes. No style changes and placed in a way that conforms with overall feel of current card design.
 
 Changes to be committed:
	modified:   src/clock-weather-card.ts to add display and method to obtain humidity, along with config
	modified:   src/types.ts to add required types for changes
	modified:   README.md to include the new options

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +136,7 @@ time_zone: null
...
 | hide_today_section    | boolean          | **Optional** | Hides the cards today section (upper section), containing the large weather icon, clock and current date                                                                                                                          | `false`   |
 | hide_forecast_section | boolean          | **Optional** | Hides the cards forecast section (lower section),containing the weather forecast                                                                                                                                                  | `false`   |
+| hide_humidity             | boolean          | **Optional** | Hides the humidity from the today section                                                                                                                                                                                             | `false`   |
```
Comment (pkissling) [inline README.md:139]: Please set the default value to `true`. i don't think it makes sense to enable humidity by default for every user

File: src/clock-weather-card.ts [hunk 2]
```diff
@@ -223,6 +226,7 @@ export class ClockWeatherCard extends LitElement {
         <clock-weather-card-today-right-wrap>
           <clock-weather-card-today-right-wrap-top>
             ${this.config.hide_clock ? weatherString : localizedTemp ? `${weatherString}, ${localizedTemp}` : weatherString}
+            ${!this.config.hide_humidity && localizedHumidity ? html`<div>${localizedHumidity} Humidity</div>` : ''}
```
Comment (pkissling) [inline src/clock-weather-card.ts:229]: do we actually require the `div` at this point?

Comment (pkissling) [inline src/clock-weather-card.ts:229]: "Humidity" must be fetched from translations

File: src/clock-weather-card.ts [hunk 3]
```diff
@@ -452,6 +458,21 @@ export class ClockWeatherCard extends LitElement {
     return this.getWeather().attributes.temperature ?? null
   }
 
+  private getCurrentHumidity (): number | null {
+    if (this.config.humidity_sensor) {
+      const humiditeeSensor = this.hass.states[this.config.humidity_sensor] as HumiditySensor | undefined;
```
Comment (pkissling) [inline src/clock-weather-card.ts:463]: `humiditeeSensor` typo

**Review by pkissling — CHANGES_REQUESTED** [review]

File: src/clock-weather-card.ts (hunk 2 above)
Comment (rakasha681) [reply src/clock-weather-card.ts]: I'm horrible with HTML, and this is my first time using TS also... so... probably not :D  How about a <br>? was just looking for a way to get it to the next line so it wouldn't clip on smaller dashboards.

Comment (rakasha681) [reply src/clock-weather-card.ts]: didn't realize what I was doing there, but I get it now. I fully localized it now.

File: src/clock-weather-card.ts (hunk 3 above)
Comment (rakasha681) [reply src/clock-weather-card.ts:465]: there is an existing interface HumiditySensor in types, and i was using your getCurrentTemperature as a template, which used 'temperatueSensor' for the same reason i think, existing TemperatureSensor, so i just went with that :)

File: README.md [hunk 4]
```diff
@@ -126,7 +126,7 @@ time_zone: null
...
 | sun_entity            | boolean          | **Optional** | ID of the sun entity. Used to determine whether to show a day or night icon. If sun integration is not enabled, day icon will be shown                                                                                            | `sun.sun` |
 | temperature_sensor    | string           | **Optional** | ID of the temperature sensor entity. Used to show the current temperature based on a sensor value instead of the weather forecast                                                                                                 | `''`      |
-| humidity_sensor    | string           | **Optional** | ID of the humidity sensor entity. Used to show the current humidity based on a sensor value                                                                                                 | `''`      |
+| humidity_sensor    | string           | **Optional** | ID of the humidity sensor entity. Used to show the current humidity based on a sensor value                                                                                                 | `true`      |
```
[the remark this answers is not in the document]
Comment (rakasha681) [reply README.md]: all set!

File: src/clock-weather-card.ts [hunk 5]
```diff
@@ -216,7 +216,9 @@ export class ClockWeatherCard extends LitElement {
     const icon = this.toIcon(state, iconType, false, this.getIconAnimationKind())
     const weatherString = this.localize(`weather.${state}`)
     const localizedTemp = temp !== null ? this.toConfiguredTempWithUnit(tempUnit, temp) : null
-    const localizedHumidity = humidity !== null ? `${humidity}%` : null;
+    const localizedHumidity = humidity !== null ? `${humidity}% ${this.localize('misc.Humidity')}` : null;
```
[the remark this answers is not in the document]
Comment (rakasha681) [reply src/clock-weather-card.ts]: all set!

**Comment by pkissling** [comment]
all good, thanks for the PR. please fix the lint issues (https://github.com/pkissling/clock-weather-card/actions/runs/8253081526/job/22601291054?pr=337), so that i can merge the changes and create a release.

one thing which i am thinking about, is that the default `hide_humidity: true` should become `show_humidity: false`, otherwise it would imply that the humidity is shown by default, which is not true. what do you think?

**Comment by rakasha681** [comment]
I was thinking the same thing. Changed that, and fixed the lint. Glad to help out, not too bad for my first PR ever, and first time using TS :)

**Review by pkissling — APPROVED** [review]

**Pull request merged**